### PR TITLE
Reject invalid function calls

### LINKED ISSUE #6: Compile-time errors related to `this` are gone
Let's suppose I have a function that uses `this` other than `Window`. In this example, I have a static method that _must_ be called with `this` of `JSON`.

```ts
declare class JSON {
  /**
   * `parse` should always be called with `this === JSON`.
   */
  static parse(this: JSON, input: string): object;
}

const { parse } = JSON;

JSON.parse('1'); // Good
parse('1') // Bad, we get a compile-time error as expected
```

TypeScript will warn you when you try and call `parse` as if it were a standalone function.

However, when we do the same thing with `pipe`/`pipeWith`, the error is gone.

```ts
pipeWith(
  'foo',
  JSON.parse // Good: `parse` is called with `this` of JSON
)

pipeWith(
  'foo',
  parse // Bad: this should give us a compile-time error!
)
```

### PR DESCRIPTION
**Pull request by karol-majewski**
Closes #6.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,24 @@ assert.strictEqual(result, 4);
...
+**Bad**
+
+```ts
+pipeWith('foo', JSON.parse);
+```
+
+**Good**
+
+```ts
+pipeWith('foo', input => JSON.parse(input));
```
Comment (OliverJAsh) [inline README.md]: Or `JSON.parse.bind(JSON)`?

Comment (karol-majewski) [reply README.md]: Correct!

(I changed the example to `localStorage` instead of `JSON` since JSON seems to be working with any `this` after all.)

**Comment by OliverJAsh** [comment]
@karol-majewski Are you able to do a release and then bump our version?

**Comment by karol-majewski** [comment]
@OliverJAsh Published as v0.0.8.

**Pull request merged**